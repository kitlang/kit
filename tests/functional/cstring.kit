function main() {
    for i in "hello, world!" {
        printf("%c", i);
    }
    puts("");

    puts(if "abcdefg".startsWith("abc") then "abcdefg starts with abc" else "wrong");
    puts(if "abcdefg".startsWith("efg") then "wrong" else "abcdefg does not start with efg");
    puts(if "abcdefg".startsWith("") then "abcdefg starts with empty string" else "wrong");
    puts(if "abcdefg".startsWith("abcdefg") then "abcdefg starts with abcdefg" else "wrong");

    puts(if "abcdefg".endsWith("efg") then "abcdefg ends with efg" else "wrong");
    puts(if "abcdefg".endsWith("def") then "wrong" else "abcdefg does not end with def");
    puts(if "abcdefg".endsWith("") then "abcdefg ends with empty string" else "wrong");
}
