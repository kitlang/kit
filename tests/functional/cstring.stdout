hello, world!
abcdefg starts with abc
abcdefg does not start with efg
abcdefg starts with empty string
abcdefg starts with abcdefg
abcdefg ends with efg
abcdefg does not end with def
abcdefg ends with empty string
